src/ToricVarieties/CohomologyClasses/constructors.jl

```diff
@@ -0,0 +1,297 @@
+#############################
+# 1: The Julia type for toric cohomology classes
+#############################
+
+@attributes mutable struct CohomologyClass
+    toric_variety::AbstractNormalToricVariety
+    coeffs::Vector
+    exponents::fmpz_mat
+    function CohomologyClass(v::AbstractNormalToricVariety, p::MPolyQuoElem)
+        if length(coordinate_names(v)) != ngens(parent(p))
+            throw(ArgumentError("The polynomial must reside in the cohomology ring of the toric variety."))
+        end
+        for k in 1:length(coordinate_names(v))
+            if string(gens(parent(p))[k]) !== coordinate_names(v)[k]
+                throw(ArgumentError("The names of the indeterminates must match the names of the homogeneous coordinates of the toric variety."))
+            end
+        end
+        coeffs = [k for k in coefficients(p.f)]
+        expos = matrix(ZZ,[k for k in exponent_vectors(p.f)])
+        if (nrows(expos) !== length(coeffs))
+            throw(ArgumentError("There must be as many exponents as coefficients."))
+        end
+        if ((iszero(p) == false) && (ncols(expos) !== ngens(cohomology_ring(v))))
```

Comment:
It's unusual to compare integers with `!==` or `===` (it's allowed, of course, just unusual; anyway, they produce the exact same machine code as `!=`, resp `==`)
```suggestion
        if !iszero(p) && ncols(expos) != ngens(cohomology_ring(v))
```

Reply:
Done. Thx.